### ISSUE TITLE
`model.run_with_cache` errors when `hook_embed` does not pass the `NameFilter`

### ISSUE THREAD

**Issue by ArthurConmy** [neelnanda-io/Easy-Transformer#45]
By default, lines such as 

`
logits, cache = model.run_with_cache(
        "Hello, Arthur",
        names_filter=lambda name: name in ["hook_pos_embed"]
)
`

error since the following line requires `hook_embed` to be cached. Is this intended? It's a bit annoying editing all my NameFilters.

https://github.com/neelnanda-io/Easy-Transformer/blob/57ee0b8eee09e3d283170702afc052b5ad8643c6/easy_transformer/activation_cache.py#L34

**Comment by neelnanda-io** [comment]
(Being fixed in a PR I'm currently writing, definitely not intentional! Thanks for flagging :) )

**Comment by neelnanda-io** [comment]
Fixed, I think!
